packages/widget/src/pages/SettingsPage/ResetSettingsButton.tsx

```diff
@@ -35,25 +44,43 @@ export const ResetSettingsButton: React.FC = () => {
     toggleDialog();
   };
 
+  const isCustomRouteSettings =
+    tools?.bridges?.length !== enabledBridges?.length ||
+    tools?.exchanges?.length !== enabledExchanges?.length;
+
   return (
     <Box px={3} mt={1.5} mb={1}>
-      <Button onClick={toggleDialog} fullWidth>
-        {t('button.resetSettings')}
-      </Button>
-      <Dialog open={open} onClose={toggleDialog}>
-        <DialogTitle>{t('warning.title.resetSettings')}</DialogTitle>
-        <DialogContent>
-          <DialogContentText>
-            {t('warning.message.resetSettings')}
-          </DialogContentText>
-        </DialogContent>
-        <DialogActions>
-          <Button onClick={toggleDialog}>{t('button.cancel')}</Button>
-          <Button variant="contained" onClick={handleReset} autoFocus>
-            {t('button.reset')}
-          </Button>
-        </DialogActions>
-      </Dialog>
+      <ResetButtonContainer isCustomRouteSettings={isCustomRouteSettings}>
+        {isCustomRouteSettings && (
+          <Box display={'flex'} marginBottom={'12px'}>
+            <InfoRounded
+              sx={{
+                marginRight: '8px',
+              }}
+            />
+            <Box marginTop={'2px'} fontSize={14}>
+              {t(`settings.resetSettings`)}
+            </Box>
+          </Box>
+        )}
+        <Button onClick={toggleDialog} fullWidth>
+          {t('button.resetSettings')}
+        </Button>
+        <Dialog open={open} onClose={toggleDialog}>
+          <DialogTitle>{t('warning.title.resetSettings')}</DialogTitle>
+          <DialogContent>
+            <DialogContentText>
+              {t('warning.message.resetSettings')}
+            </DialogContentText>
+          </DialogContent>
+          <DialogActions>
+            <Button onClick={toggleDialog}>{t('button.cancel')}</Button>
+            <Button variant="contained" onClick={handleReset} autoFocus>
+              {t('button.reset')}
+            </Button>
+          </DialogActions>
+        </Dialog>
+      </ResetButtonContainer>
```

Comment:
We should show message **and** button only if some settings have changed

Reply:
Okay, the button should be shown for any settings change whereas the message only for route affecting changes right?

Reply:
<img width="366" alt="image" src="https://github.com/lifinance/widget/assets/18644653/1d0f1358-d375-4515-9890-fff677e9cea8">

All these settings affect routes, so message and button should be shown when we change them. There should be no cases when the button is shown without a message. We can just ignore language and appearance here.

Reply:
Correct, in case of any default widget past configuration modification, I will be reverting back to the configuration passed via the widget